Source/WebCore/editing/Editor.cpp

```diff
@@ -3679,6 +3679,23 @@ RefPtr<TextPlaceholderElement> Editor::insertTextPlaceholder(const IntSize& size
 
     document->selection().setSelection(VisibleSelection { positionInParentBeforeNode(placeholder.ptr()) }, FrameSelection::defaultSetSelectionOptions(UserTriggered::Yes));
 
+    // For Writing Tools, we need the snapshot of the last inserted placeholder.
```

Comment:
Since this is just for Writing Tools, I think we should move this logic into the controller.